### ISSUE TITLE
RFC: Add ability to create, run and save SQL queries in the Admin UI

### ISSUE THREAD

**Issue by joemcl** [ProgressiveCoders/Spoke#1641]
## Proposal
Add the ability to create, run and save SQL queries in the Admin UI

## Motivation
A number of Spoke user orgs use external dashboard apps like redash.io or Periscope to create, run and save SQL queries. Adding the ability to create, run and save SQL queries right in the admin UI could save money for orgs that don't have budget at the moment for external dashboarding apps. Being able to see some basic SQL queries results in a dashboard in the admin UI could allow orgs to share that info with a wider audience of admins more quickly, and save admin users time by not having to flip back and forth between Spoke and an external dashboard.   

## Detailed Design
Thoughts appreciated :) 

#### Sponsor Organization (will test alphas/betas)
I'm happy to help test this on behalf of WFP

#### Developer allocated
Noone particular as of yet

#### Planned timeline
This would be handy to have in place well before November 2020

#### Schema changes and migrations
??

**Comment by arena** [comment]
I can mock this up if somebody wants to build it

**Comment by ibrand** [comment]
@joemcl were you imagining that this would render in approximately the same way it would render if you did the sql query in periscope? Like plain text in columns?

I think this would be an interesting module for an admin tab meant for an org-level overview of "quick stats" for the last few days. Like to see how many texts were sent across campaigns

**Comment by joemcl** [comment]
@ibrand yeah, render as simple tables as a start. Then it would be handy to be able to add data visualizations down the road.
